packages/core/lib/src/core_html_widget.dart

```diff
@@ -16,51 +16,51 @@ import 'internal/tsh_widget.dart';
 /// (supports most popular tags and stylings).
 class HtmlWidget extends StatefulWidget {
   /// The base url to resolve links and image urls.
-  final Uri baseUrl;
+  final Uri? baseUrl;
 
   /// Controls whether the widget tree is built asynchronously.
   ///
   /// If not set, async build will be enabled automatically if the
   /// [html] has at least [kShouldBuildAsync] characters.
-  final bool buildAsync;
+  final bool? buildAsync;
 
   /// The callback to handle async build snapshot.
   ///
   /// By default, a [CircularProgressIndicator] will be shown until
   /// the widget tree is ready.
   /// This default builder doesn't do any error handling
   /// (it will just ignore any errors).
-  final AsyncWidgetBuilder<Widget> buildAsyncBuilder;
+  final AsyncWidgetBuilder<Widget>? buildAsyncBuilder;
 
   /// The callback to specify custom stylings.
-  final CustomStylesBuilder customStylesBuilder;
+  final CustomStylesBuilder? customStylesBuilder;
 
   /// The callback to render a custom widget.
-  final CustomWidgetBuilder customWidgetBuilder;
+  final CustomWidgetBuilder? customWidgetBuilder;
 
   /// Controls whether the built widget tree is cached between rebuilds.
   ///
   /// Default: `true` if [buildAsync] is off, `false` otherwise.
-  final bool enableCaching;
+  final bool? enableCaching;
```

Comment:
```suggestion
  final bool enableCaching;
```

Reply:
OK, made it not nullable (along with buildAsync)

Reply:
It looks like this is not easily possible... I tried to make it `late` and calculate the default in the constructor, but that somehow caused a test to fail. I am not sure why, will have another look.

Reply:
Hmm, let keep it optional then.